### ISSUE TITLE
Add an option to check for misspellings in special functions

### ISSUE THREAD

**Issue by kunaltyagi** [PyCQA/pylint#3038]
### Is your feature request related to a problem? Please describe
Currently, the errors in naming of special functions aren't checked. Normal functions act as interfaces and they are readily checked. But for special functions, it becomes hard to confirm that the code written will work and many times, the eyes gloss over things like number of underscores (too many or too few, hard to distinguish), spelling (should that be invert or inv, eq or equal or equals).
```py
class abc:
    __init_(self):
        pass
    ___neg__(self):
        pass
    __inv__(self):
        pass
```

### Describe the solution you'd like
Since the special functions are limited, testing for spellings (and offering the suggestions as a plugin) would be ideal.

**Comment by PCManticore** [comment]
This sounds useful, thanks for the suggestion!

**Comment by kunaltyagi** [comment]
Any tips on how to add this feature?

**Comment by Pierre-Sassoulas** [comment]
I guess it's possible to use that if a function name start and ends with ``__`` then it's one of the official dunder methods for python.
